homeassistant/components/ialarmxr/config_flow.py

```diff
@@ -0,0 +1,68 @@
+"""Config flow for Antifurto365 iAlarmXR integration."""
+import logging
+
+from pyialarmxr import IAlarmXR
+import voluptuous as vol
+
+from homeassistant import config_entries, core
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_PORT, CONF_USERNAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DEFAULT_HOST, DEFAULT_PORT, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST, default=DEFAULT_HOST): str,
+        vol.Required(CONF_PORT, default=DEFAULT_PORT): int,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+async def _get_device_mac(
+    hass: core.HomeAssistant, host: str, username: str, password: str, port: int
+) -> str:
+    ialarm = IAlarmXR(host, username, password, port)
+    return await hass.async_add_executor_job(ialarm.get_mac)
+
+
+class IAlarmConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Antifurto365 iAlarmXR."""
+
+    VERSION = 1
+
+    async def async_step_user(self, user_input=None) -> FlowResult:
+        """Handle the initial step."""
+        errors = {}
+        mac = None
+
+        if user_input is None:
+            return self.async_show_form(step_id="user", data_schema=DATA_SCHEMA)
+
+        host = user_input[CONF_HOST]
+        port = user_input[CONF_PORT]
+        username = user_input[CONF_USERNAME]
+        password = user_input[CONF_PASSWORD]
+
+        try:
+            # If we are able to get the MAC address, we are able to establish
+            # a connection to the device.
+            mac = await _get_device_mac(self.hass, host, username, password, port)
```

Comment:
Will there be another exception when the username or password is incorrect? (but a connection is possible).

Reply:
ConnectionError will be thrown if host and/or port will be unreachable but if login and/or password will be incorrect then a generic error will be thrown.